pkg/controller/controller.go

```diff
@@ -980,29 +1032,71 @@ func removePrefixedParameters(param map[string]string) (map[string]string, error
 // so that an external controller can act upon it.   Additional DataSource types can be added here with
 // an appropriate implementation function
 func (p *csiProvisioner) getVolumeContentSource(ctx context.Context, claim *v1.PersistentVolumeClaim, sc *storagev1.StorageClass) (*csi.VolumeContentSource, error) {
-	switch claim.Spec.DataSource.Kind {
-	case snapshotKind:
-		return p.getSnapshotSource(ctx, claim, sc)
-	case pvcKind:
-		return p.getPVCSource(ctx, claim, sc)
-	default:
-		// For now we shouldn't pass other things to this function, but treat it as a noop and extend as needed
-		return nil, nil
+	if claim.Spec.DataSourceRef != nil {
+		switch claim.Spec.DataSourceRef.Kind {
+		case snapshotKind:
+			return p.getSnapshotSource(ctx, claim, sc)
+		case pvcKind:
+			return p.getPVCSource(ctx, claim, sc)
+		default:
+			// For now we shouldn't pass other things to this function, but treat it as a noop and extend as needed
+			return nil, nil
+		}
+	} else {
```

Comment:
Do we need the if/else? The logic looks exactly the same.

Reply:
Fixed.